das/daser_test.go

```diff
@@ -0,0 +1,52 @@
+package das
+
+import (
+	"context"
+	"sync"
+	"testing"
+
+	"github.com/celestiaorg/celestia-node/service/header"
+	"github.com/celestiaorg/celestia-node/service/share"
+)
+
+func TestDASer(t *testing.T) {
+	shareServ, dah := share.RandServiceWithSquare(t, 16)
+
+	randHeader := header.RandExtendedHeader(t)
+	randHeader.DataHash = dah.Hash()
+	randHeader.DAH = dah
+
+	sub := &mockHeaderSub{
+		headers: []*header.ExtendedHeader{randHeader},
+	}
+
+	daser := NewDASer(shareServ, sub)
+
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+	go func(wg *sync.WaitGroup) {
+		daser.sampling(context.Background(), sub)
```

Comment:
The reason I was really lazy about writing the test here is that I didn't want to make a mock Subscriber which returns a mock subscription. Your solution to the problem is great!